src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,57 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    private static final String CAN_T_ADD_NEW_USER_MSG = "Can't add new user";
+    private static final String SPACE = " ";
+    private static final String EMAIL = "email";
+    private static final String CAN_T_RECEIVE_USER_BY_EMAIL_MSG = "Can't receive user by email ";
+
+    @Override
+    public User add(User user) {
+        Session session = null;
```

Comment:
The `session` variable is declared outside the try block, which is not necessary. You can declare it inside the try block to limit its scope and improve code readability.